docs/iscsi-authentication.md

```diff
@@ -0,0 +1,71 @@
+This document describes how to associate a k8s secret with a VM for the purpose of iscsi initiator authentication.
+
+*NOTE: Only client authentication is supported at this time, meaning that the iscsi target with authenticate an initiator has permissions to access a device, but that a initiator can not authenticate the target.*
+
+## Workflow
+1. create a k8s secret containing the password and username fields.
+```
+cat << END > my-chap-secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: myK8sSecretID
+  namespace: default
+
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-chap-secret
+type: "kubernetes.io/iscsi-chap"  
+data:
+  node.session.auth.username: $(echo "myUsername" | base64 -w0)
+  node.session.auth.password: $(echo "mySuperSecretPassword" | base64 -w0)
+END
+```
+2. create a vm that references the **name** given to the k8s secret in the iscsi usage field
+```
+cat << END > my-vm.yaml
+kind: VM
+metadata:
+  name: testvm
+  namespace: default
+spec:
+  domain:
+    devices:
+      disks:
+      - Auth:
+          secret:
+            type: iscsi
+            usage: my-chap-secret
+        type: network
+        snapshot: external
+        device: disk
+        driver:
+          name: qemu
+          type: raw
+          cache: none
+        source:
```

Comment:
iscsi persistent volume is just a wrapper around the options available to an iscsi volume. I verified this in the k8s code base. 

The example below is how iscsi auth should look like with a PVC.

In this example, the user would create the chap-secret k8s secret and then define this PV which references the secret. 

```
apiVersion: v1
kind: PersistentVolume
metadata:
  name: iscsi-disk-custom
  labels:
    os: "none"
spec:
  capacity:
    storage: 1Gi
  accessModes:
    - ReadWriteOnce
  iscsi:
    iqn: iqn.2017-01.io.kubevirt:sn.42
    lun: 1
    targetPortal: iscsi-demo-target.default.svc.cluster.local
    chapAuthDiscovery: true
    chapAuthSession: true
    secretRef: 
       name: chap-secret
```